### PR TITLE
Remove lower server rankings for unknown maps

### PR DESCRIPTION
**Pull request by WilliamVenner**
:+1:

We should be promoting new and unique maps.

### CODE REVIEW

**Comment by willox** [comment]
This should probably be merged just because it doesn't count previously cached workshop addons

**Comment by aStonedPenguin** [comment]
please

**Comment by MonkeysLP** [comment]
+1

**Comment by zeaga** [comment]
+1

**Comment by roastt** [comment]
+1

**Comment by KingofBeast** [comment]
:+1:

**Comment by dr-alto-clef** [comment]
+1

**Comment by Kefta** [comment]
Good PR

**Comment by Zaddion** [comment]
+1

**Comment by BillyOnWiiUIn144p** [comment]
Thank you, I am always happy to contribute to Garry's Mod with such impacting PRs

**Comment by Brandino480** [comment]
+1, You gotta keep Garrys Mod alive, Promoting new user content will certainly extend the games life span atleast another 5 years.

**Comment by Blueasharky** [comment]
+1 :+1:   #MakeGModGreatAgain

**Comment by xognovalis** [comment]
+1

**Comment by weard1212** [comment]
+1

**Comment by fruitwasp** [comment]
+1

File: garrysmod/html/js/menu/control.Servers.js [hunk 1]
```diff
@@ -233,7 +233,6 @@ function AddServer( type, id, ping, name, desc, map, players, maxplayers, botpla
 	data.hasmap = DoWeHaveMap( data.map );
```
Comment (bigdogmat) [inline garrysmod/html/js/menu/control.Servers.js]: Might as well remove this if it's not being used elsewhere

Comment (BillyOnWiiUIn144p) [inline garrysmod/html/js/menu/control.Servers.js]: @bigdogmat https://github.com/BillyOnWiiUIn144p/garrysmod/blob/master/garrysmod/html/js/menu/control.Servers.js#L83

**Comment by mcNuggets1** [comment]
+1

File: garrysmod/html/js/menu/control.Servers.js [hunk 2]
```diff
@@ -230,10 +230,7 @@ function AddServer( type, id, ping, name, desc, map, players, maxplayers, botpla
 		workshopid:		workshopid
 	};
 
-	data.hasmap = DoWeHaveMap( data.map );
```
Comment (Bo98) [inline garrysmod/html/js/menu/control.Servers.js]: This is used in `$scope.ServerClass`.

Comment (WilliamVenner) [inline garrysmod/html/js/menu/control.Servers.js]: Are you sure? It's data.hasmap and at a first glance, it shows that the sv variable isn't the data variable but I could be wrong.

Comment (Bo98) [inline garrysmod/html/js/menu/control.Servers.js]: `sv` and `data` aren't global variables. This data object is added to the `servers` array in the gamemode object. When being looped over, `ServerClass` is called, with this data object passed in as a parameter.

Best thing is to test it and see if the missing maps are coloured red or not.

**Comment by thegrb93** [comment]
+1000

**Comment by OneEaredNeko** [comment]
+1

**Comment by robotboy655** [comment]
https://github.com/garrynewman/garrysmod/commit/b3171137890829e2322e9c564b0d0baae9827255

**Pull request closed without merging**